packages/pdf-viewer/VerticalPages.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useLayoutEffect, useRef, useState } from 'react';
+import { PdfData, ScaledSize } from './pdfSource';
+import Page from './Page';
+
+require('./pages.css');
+
+export default function VerticalPages(props: {
+	pdf: PdfData;
+	isDarkTheme: boolean;
+	anchorPage: number;
+	container: React.MutableRefObject<HTMLElement>;
+}) {
+	const [scaledSize, setScaledSize] = useState<ScaledSize>(null);
+	const innerContainerEl = useRef<HTMLDivElement>(null);
+	const resizeTimer = useRef<number>();
+	useLayoutEffect(() => {
+		const updateSize = async () => {
```

Comment:
Add a "let cancelled = false" at the top. On exit, set it to "true".

Then everywhere you have async code you need to handle it. `if (cancelled) return`. Otherwise you might be running code when the component has been dismounted.

This is a common pattern in React Hooks, please google it.